bridge/device/credential/manager.go

```diff
@@ -0,0 +1,172 @@
+/****************************************************************************
+ *
+ * Copyright (c) 2024 plgd.dev s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+package credential
+
+import (
+	"crypto/x509"
+
+	"github.com/plgd-dev/device/v2/bridge/net"
+	"github.com/plgd-dev/device/v2/bridge/resources"
+	"github.com/plgd-dev/device/v2/pkg/codec/cbor"
+	"github.com/plgd-dev/device/v2/schema/credential"
+	"github.com/plgd-dev/device/v2/schema/interfaces"
+	"github.com/plgd-dev/go-coap/v3/message/codes"
+	"github.com/plgd-dev/go-coap/v3/message/pool"
+	"github.com/plgd-dev/go-coap/v3/pkg/sync"
+	"github.com/plgd-dev/kit/v2/security"
+)
+
+type Manager struct {
+	credentials *sync.Map[int, credential.Credential]
+	save        func()
+}
+
+func New(save func()) *Manager {
+	if save == nil {
+		save = func() {
+			// do nothing
+		}
+	}
+	return &Manager{
+		save:        save,
+		credentials: sync.NewMap[int, credential.Credential](),
+	}
+}
+
+func (m *Manager) GetCAPool() []*x509.Certificate {
+	certs := make([]*x509.Certificate, 0, m.credentials.Length())
+	m.credentials.Range(func(key int, value credential.Credential) bool {
+		if value.Type != credential.CredentialType_ASYMMETRIC_SIGNING_WITH_CERTIFICATE {
+			return true
+		}
+		if value.Usage != credential.CredentialUsage_TRUST_CA && value.Usage != credential.CredentialUsage_MFG_TRUST_CA {
+			return true
+		}
+		cas, err := security.ParseX509FromPEM(value.PublicData.Data())
```

Comment:
equivalent function should be available in pkg/x509

Reply:
fixed